File: frontend/lib/models/Paciente.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface IPaciente extends Document {
  nome: string;
  cpf: string;
  dataNascimento: string;
  sexo: string;
  email: string;
  senha?: string; // Hashed
  telefone?: string;
  endereco?: string;
  tipoSanguineo?: string;
  doadorOrgaos?: boolean;
  alergias?: string[];
  doencasCronicas?: string[];
  medicamentosContinuos?: {
    nome: string;
    dosagem: string;
    horario: string;
  }[];
}

const PacienteSchema: Schema = new Schema({
  nome: { type: String, required: true },
  cpf: { type: String, required: true, unique: true },
  dataNascimento: { type: String, required: true },
  sexo: { type: String, required: true },
  email: { type: String, required: true, unique: true },
  senha: { type: String, required: true },
  telefone: { type: String },
  endereco: { type: String },
  tipoSanguineo: { type: String },
  doadorOrgaos: { type: Boolean, default: false },
  alergias: [{ type: String }],
  doencasCronicas: [{ type: String }],
  medicamentosContinuos: [{
    nome: { type: String },
    dosagem: { type: String },
    horario: { type: String }
  }]
}, {
  timestamps: true,
  collection: 'Pacientes'
});

export default mongoose.models.Paciente || mongoose.model<IPaciente>('Paciente', PacienteSchema);

File: backend/SaudeMemora.Domain/Entities/Paciente.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace SaudeMemora.Domain.Entities;

public class MedicamentoContinuo
{
    [BsonElement("nome")]
    public string Nome { get; set; } = string.Empty;

    [BsonElement("dosagem")]
    public string Dosagem { get; set; } = string.Empty;

    [BsonElement("horario")]
    public string Horario { get; set; } = string.Empty;
}

public class Paciente
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string? Id { get; set; }

    [BsonElement("nome")]
    public string Nome { get; set; } = string.Empty;

    [BsonElement("cpf")]
    public string Cpf { get; set; } = string.Empty;

    [BsonElement("dataNascimento")]
    public string DataNascimento { get; set; } = string.Empty;

    [BsonElement("sexo")]
    public string Sexo { get; set; } = string.Empty;

    [BsonElement("email")]
    public string Email { get; set; } = string.Empty;

    [BsonElement("senha")]
    public string Senha { get; set; } = string.Empty;

    [BsonElement("telefone")]
    public string? Telefone { get; set; }

    [BsonElement("endereco")]
    public string? Endereco { get; set; }

    [BsonElement("tipoSanguineo")]
    public string? TipoSanguineo { get; set; }

    [BsonElement("doadorOrgaos")]
    public bool? DoadorOrgaos { get; set; }

    [BsonElement("alergias")]
    public List<string> Alergias { get; set; } = new();

    [BsonElement("doencasCronicas")]
    public List<string> DoencasCronicas { get; set; } = new();

    [BsonElement("medicamentosContinuos")]
    public List<MedicamentoContinuo> MedicamentosContinuos { get; set; } = new();

    [BsonElement("createdAt")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
